flocker/node/agents/functional/test_cinder.py

```diff
@@ -335,7 +329,9 @@ def test_get_device_path_correct_with_attached_disk(self):
         instance_id = self.blockdevice_api.compute_instance_id()
 
         host_device = "/dev/null"
-        virtio = VirtIOClient.from_instance_id(instance_id)
+        tmpdir = FilePath(self.mktemp())
+        tmpdir.makedirs()
+        virtio = VirtIOClient.using_insecure_tls(instance_id, tmpdir)
```

Comment:
:question: It's a matter of opinion, but I think if I was doing this, I might: 
 * make `create_credentials` a free function (rather than a static method) which returns the `FilePath` 
 * then use it as follows: 
```
virtio = VirtIOClient.using_insecure_tls(
    instance_id=instance_id, 
    credentials_directory=create_credentials(
        base_directory=FilePath(self.mktemp())
    )
) 
```